tools/modify-image

```diff
@@ -0,0 +1,222 @@
+      '{Path: $path, NewHash: $nh, OldHash: $oh}' )
+
+  CHANGED_ROOTS+=(${JSON_STRING})
+}
+
+for opt in "$@"; do
+   optarg="$(expr "${opt}" : '[^=]*=\(.*\)')"
+   case "${opt}" in
+      --image=*) IMAGE_PATH="${optarg}" ;;
+      --image-out=*) OUT_IMAGE="${optarg}" ;;
+      --new-root=*) ROOTFILE="${optarg}" ;;
+      --ca-bundle=*) CABUNDLE="${optarg}" ;;
+   esac
+done
+
+echo -e "here are the vars: ${IMAGE_PATH} ${OUT_IMAGE} ${ROOTFILE} ${CABUNDLE}"
+
+DESTDIR=$(dirname ${IMAGE_PATH})
+WORKDIR="$(mktemp -d)"
+cleanup() {
+   [ -n "${WORKDIR}" ] && rm -rf "${WORKDIR}"
+}
+trap 'cleanup' EXIT
+
+cp ${IMAGE_PATH} ${WORKDIR}
+
+pushd "${WORKDIR}" >/dev/null
+
+IMAGE=$(basename "${IMAGE_PATH}")
+
+if [[ ${IMAGE} == *.lz4 ]] ; then
+echo "Extracting ${IMAGE}"
+  rm -f ${IMAGE%.*}
+  unlz4 ${IMAGE}
+  IMAGE=${IMAGE%.*}
+fi
+
+IMAGE_NAME=${IMAGE%.*}
+# step 2: extract the root filesystem
+echo -e "Starting extract of root for ${IMAGE}"
+ROOT_IMAGE="root.ext4"
+ROOT_START="$(sgdisk --print "${IMAGE}" | awk '/BOTTLEROCKET-ROOT-A/{ print $2 }')"
+echo ${ROOT_START}
+ROOT_END="$(sgdisk --print "${IMAGE}" | awk '/BOTTLEROCKET-ROOT-A/{ print $3 }')"
+echo ${ROOT_END}
+ROOT_SIZE="$(( ROOT_END - ROOT_START + 1))"
+dd if="${IMAGE}" of="${ROOT_IMAGE}" skip="${ROOT_START}" count="${ROOT_SIZE}"
+
+echo "File exists at ${ROOT_IMAGE}"
+
+# step 3: replace the trusted root.json
+# os_t is the right label for SELinux based upon defaults
+if [ -n "${ROOTFILE}" ] ; then
+  declare -A ROOT_ARR
+  JSON_STRING=""
+  ROOT_ARR[Path]='/usr/share/updog/root.json'
+  ROOT_ARR[OldHash]=$(debugfs -f <(echo cat /usr/share/updog/root.json) ${ROOT_IMAGE} 2>/dev/null | tail -n +2 | sha512sum | head -c 128)
+  ROOT_ARR[NewHash]=$(sha512sum ${ROOTFILE} | head -c 128)
+  buildJson ROOT_ARR
+cat <<EOF | debugfs -w -f - "${ROOT_IMAGE}"
+rm /usr/share/updog/root.json
+write ${ROOTFILE} /usr/share/updog/root.json
+ea_set /usr/share/updog/root.json  security.selinux system_u:object_r:os_t:s0
+EOF
+fi
+
+# replace CA Bundle if requested
+if [ -n "${CABUNDLE}" ] ; then
+declare -A CA_ARR
+JSON_STRING=""
+CA_ARR[Path]='/usr/share/factory/etc/pki/tls/certs/ca-bundle.crt'
+CA_ARR[OldHash]=$(debugfs -f <(echo cat /usr/share/factory/etc/pki/tls/certs/ca-bundle.crt) ${ROOT_IMAGE} 2>/dev/null | tail -n +2 | sha512sum | head -c 128)
+CA_ARR[NewHash]=$(sha512sum ${CABUNDLE} | head -c 128)
+buildJson CA_ARR
+cat <<EOF | debugfs -w -f - "${ROOT_IMAGE}"
+rm /usr/share/factory/etc/pki/tls/certs/ca-bundle.crt
+write ${CABUNDLE} /usr/share/factory/etc/pki/tls/certs/ca-bundle.crt
+ea_set /usr/share/factory/etc/pki/tls/certs/ca-bundle.crt security.selinux system_u:object_r:os_t:s0
+EOF
+fi
+
+# Create the full manifest of modifications for writing to trust-roots.json
+echo -e "The full list of json is: ${CHANGED_ROOTS[@]}"
+MODIFIED_JSON="$(jq --raw-output . <<<  "${CHANGED_ROOTS[@]}")"
+
+FULL_MODIFIED_JSON="$(jq --slurp 'sort_by(.Name)' <<< "${MODIFIED_JSON}" | jq '{"ModifiedRootsOfTrust": .}')"
```

Comment:
I have doubts about creating this manifest as a file in the image, but I could see adding this as an additional output for any newly built or modified image.